### PR TITLE
Resolve CVE-2023-47248 for huggingface-pytorch-inference-neuronx:1.13.1-transformers4.34.1-neuronx-py310-sdk2.15.0-ubuntu20.04

### PR DESCRIPTION
**Pull request by malav-shastri**
*GitHub Issue #, if available:*

**Note**: 
- If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

- All PR's are checked weekly for staleness. This PR will be closed if not updated in 30 days.

### Tests run
tests passed for [ca71f24](https://github.com/aws/deep-learning-containers/pull/3813/commits/ca71f2425b91c6e19f862edca40aed17861d3d5a)

**NOTE: By default, docker builds are disabled. In order to build your container, please update dlc_developer_config.toml and specify the framework to build in "build_frameworks"**
- [ ] I have run builds/tests on commit <INSERT COMMIT ID> for my changes.

**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**
<details>
<summary>Expand</summary>

- [ ] `sagemaker_remote_tests = true`
- [ ] `sagemaker_efa_tests = true`
- [ ] `sagemaker_rc_tests = true`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

</details>

### Formatting
- [ ] I have run `black -l 100` on my code (formatting tool: https://black.readthedocs.io/en/stable/getting_started.html)

#### Builds to Execute
<details>
<summary>Expand</summary>

Click the checkbox to enable a build to execute upon merge.

*Note: By default, pipelines are set to "latest". Replace with major.minor framework version if you do not want "latest".*

- [ ] build_pytorch_training_latest
- [ ] build_pytorch_inference_latest
- [ ] build_tensorflow_training_latest
- [ ] build_tensorflow_inference_latest

</details>

### PR Checklist 
<details>
<summary>Expand</summary>

- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

#### NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ec2_benchmark_tests = true` or `sagemaker_benchmark_tests = true`
</details>

### Pytest Marker Checklist
<details>
<summary>Expand</summary>

- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type
</details>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: dlc_developer_config.toml [hunk 1]
```diff
@@ -74,7 +74,7 @@ ec2_tests_on_heavy_instances = false
 sagemaker_local_tests = false
 
 # run standard sagemaker remote tests from test/sagemaker_tests
-sagemaker_remote_tests = false
```
Comment (jinyoung-lim) [inline dlc_developer_config.toml:77]: I think you also need to add a new buildpec (refer to the old buildspec-neuronx.yaml in its change history) and make changes under [buildspec_override] to use that buildspec to have the pipeline pick up the correct transformers version.

**Review by jinyoung-lim — CHANGES_REQUESTED** [review]
I think you also need to add a new buildpec (refer to the old buildspec-neuronx.yaml in its change history) and make changes under [buildspec_override] to use that buildspec to have the pipeline pick up the correct transformers version.

File: dlc_developer_config.toml (hunk 1 above)
Comment (malav-shastri) [reply dlc_developer_config.toml:77]: thanks JJ, I changed the buildspec file, however its still failing the checks

Comment (jinyoung-lim) [reply dlc_developer_config.toml:77]: It seems like the image is not pushed. Where is the pr hook trying to look up the pre-push image and when/where is it supposed to be built/pushed?

File: dlc_developer_config.toml [hunk 2]
```diff
@@ -74,7 +74,7 @@ ec2_tests_on_heavy_instances = false
...
+sagemaker_remote_tests = "standard"
```
Comment (jinyoung-lim) [inline dlc_developer_config.toml:77]: Also what does remote tests do? I do not see from the PR template or README.md to enable this?

Comment (malav-shastri) [reply dlc_developer_config.toml:77]: the example PR in out DLC vuln runbook talks about these tests. I'll share with you the reference offline

**Review by jinyoung-lim — APPROVED** [review]
Tests passing. Once you revert dlc-dev-config, should be good to merge.

**Review by jinyoung-lim — APPROVED** [review]

**Pull request merged**